staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -217,16 +217,16 @@ func rewriteHTML(reader io.Reader, writer io.Writer, urlRewriter func(*url.URL)
 // to the original host to instead refer to the proxy transport.
 func (t *Transport) rewriteResponse(req *http.Request, resp *http.Response) (*http.Response, error) {
 	origBody := resp.Body
-	defer origBody.Close()
```

Comment:
On supported encoding, this function reads the body, and modify the body to a rewritten one. The (modified) body is expected to be read after calling this function, to do the actual proxying, sending it to the client.
On unrecognized encoding, no rewriting can be done and this function should return the response as-is, keep the body open and unread, so it can still be read.